src/hotspot/share/cds/heapShared.cpp

```diff
@@ -873,6 +875,16 @@ void HeapShared::initialize_from_archived_subgraph(JavaThread* current, Klass* k
     return; // nothing to do
   }
 
+  ResourceMark rm(THREAD);
+  if ((strcmp(k->name()->as_C_string(), "jdk/internal/module/ArchivedModuleGraph") == 0) &&
+      !CDSConfig::is_using_optimized_module_handling() &&
+      // archive was created with --module-path
+      ClassLoaderExt::num_module_paths() > 0) {
+    log_info(cds, heap)("    is_using_optimized_module_handling %d num_module_paths %d jdk.module.main %s",
+        CDSConfig::is_using_optimized_module_handling(), ClassLoaderExt::num_module_paths(), Arguments::get_property("jdk.module.main"));
```

Comment:
Why are you printing a bool value as an int? I'm surprised one of the format checkers doesn't complain about it.

Reply:
I changed it to BOOL_TO_STR and updated the log statement.